pkg/vault/test/foundry/CompositeLiquidityRouterNestedPools.t.sol

```diff
@@ -210,21 +210,20 @@ contract CompositeLiquidityRouterNestedPoolsTest is BaseERC4626BufferTest {
         );
 
         // Check ChildPoolB balances.
-        assertApproxEqAbs(
-            vars.childPoolBAfter.dai,
-            vars.childPoolBBefore.dai + amountsIn[vars.daiIdx],
-            MAX_ROUND_ERROR,
-            "ChildPoolB Dai Balance is wrong"
-        );
+        assertEq(vars.childPoolBAfter.dai, vars.childPoolBBefore.dai, "ChildPoolB Dai Balance is wrong");
```

Comment:
Although this test adds DAI to the top-most pool, that's not always true. DAI address comes before the ChildPoolB BPT address. So, the logic is still the same as before: adds the token to the first pool that needs it that appears in the recursion. The difference is that the parent tokens are checked first.